src/tcgwars/logic/impl/gen4/SupremeVictors.groovy

```diff
@@ -262,345 +280,488 @@ public enum SupremeVictors implements LogicCardInfo {
+                  h.object += hp(40)
+                }
+              }
             }
           }
           move "Shadow Ball", {
-            text "Choose 1 of your opponent’s Benched Pokémon. This attack does 40 damage to that Pokémon. Apply Weakness and Resistance for this attack."
+            text "Choose 1 of your opponent's Benched Pokémon. This attack does 40 damage to that Pokémon. Apply Weakness and Resistance."
             energyCost P, C, C
-            attackRequirement {}
+            attackRequirement {
+              assert opp.bench : "Your opponent has no Benched Pokémon"
+            }
             onAttack {
-              damage 0
+              def pcs = opp.all.select()
+              new ResolvedDamage(hp(40), self, pcs, ATTACK, DamageManager.DamageFlag.FORCE_WEAKNESS_RESISTANCE).run(bg)
+              bg.dm().applyWeakness()
+              bg.dm().applyResistance()
+              def damage = bg.dm().getTotalDamage(self, pcs)
+              bg.dm().clearDamages()
+              bg.em().run(new DirectDamage(damage, pcs))
             }
           }
-
         };
       case ELECTIVIRE_FB_4:
-        return basic (this, hp:HP090, type:LIGHTNING, retreatCost:3) {
+        return basic (this, hp:HP090, type:L, retreatCost:3) {
           weakness F
           resistance M, MINUS20
           move "Dump and Draw", {
             text "Discard up to 2 Energy cards from your hand. For each card you discarded, draw 2 cards."
             energyCost C
-            attackRequirement {}
+            attackRequirement {
+              assert my.hand.filterByType(ENERGY) : "You have no Energy cards in your hand"
+              assert my.deck : "Your deck is empty"
+            }
             onAttack {
-              damage 0
+              def energyCards = my.hand.select(min:0,max:2,text,cardTypeFilter(ENERGY))
+              draw 2 * energyCards.size()
+              energyCards.discard()
             }
           }
           move "Electric Current", {
-            text "40 damage. to 1 of your Benched Pokémon."
-            energyCost L, C, C, L
+            text "40 damage. Move a [L] Energy card attached to Electivire FB to 1 of your Benched Pokémon."
+            energyCost L, C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 40
+              afterDamage {
+                if(my.bench && self.cards.energyCount(L)) {
+                  def energyCard = self.cards.select("Choose a [L] Energy to move",energyFilter(L)).fist()
+                  def tar = my.bench.select("Move $energyCard to which Pokémon?")
+                  energySwitch(self,tar,energyCard)
+                }
+              }
             }
           }
-
         };
       case GARCHOMP_5:
-        return evolution (this, from:"Gabite", hp:HP130, type:COLORLESS, retreatCost:0) {
-          weakness C, PLUS30
+        return evolution (this, from:"Gabite", hp:HP130, type:C, retreatCost:null) {
```

Comment:
Are you sure about the retreatCost of null here?

Reply:
This does work and is used in many of the new groovy implementations, but I agree that a cost of 0 is more readable. This is another artifact of regenerating the template.